compiler/src/dotty/tools/dotc/core/tasty/TreeUnpickler.scala

```diff
@@ -580,8 +580,9 @@ class TreeUnpickler(reader: TastyReader,
         }
       sym.annotations = annotFns.map(_(sym))
       if sym.isOpaqueAlias then sym.setFlag(Deferred)
+      val isScala2MacroDefinedInScala3 = flags.is(Macro, butNot = Inline) && flags.is(Erased)
       ctx.owner match {
-        case cls: ClassSymbol => cls.enter(sym)
+        case cls: ClassSymbol if !isScala2MacroDefinedInScala3 => cls.enter(sym)
```

Comment:
should the symbol even be created?

Reply:
We may need to load the tree and symbol for tools like the IDE and semanticdb.

Reply:
In the Scala 2 unpickler we can probably ignore the inline definition completely.